packages/web/src/common/store/tipping/selectors.ts

```diff
@@ -31,3 +40,143 @@ export const getTipToDisplay = (state: CommonState) =>
   state.tipping.tipToDisplay
 export const getShowTip = (state: CommonState) => state.tipping.showTip
 export const getMainUser = (state: CommonState) => state.tipping.mainUser
+
+const getOptimisticSupportingBase = (state: CommonState) => {
+  const { supporting, supportingOverrides } = state.tipping
+
+  /**
+   * Copy the supporting map into the eventually-merged map.
+   */
+  let mergedMap: SupportingMap = {}
+  Object.assign(mergedMap, supporting)
+
+  /**
+   * Merge the default and override maps.
+   */
+  const userIds = (Object.keys(supportingOverrides) as unknown) as ID[]
+  for (const userId of userIds) {
+    // If the supporting map for a given user id exists in the overrides
+    // but not in the default, copy the override map into the merged map.
+    const shouldOverrideMap = !supporting[userId]
+    if (shouldOverrideMap) {
+      mergedMap = {
```

Comment:
same here, I don't think we need a spread assign?

Reply:
on runtime when not using a spread assign, i got something like this
`Uncaught TypeError: Cannot assign to read only property '' of object '#<Object>`

Reply:
Hmm, would be good to not spread assign as I think that's extra expensive. We should figure out the typescript for this